src/main/java/org/datadog/jenkins/plugins/datadog/clients/ConcurrentMetricCounters.java

```diff
@@ -2,13 +2,31 @@
 
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.locks.ReentrantLock;
 
 public class ConcurrentMetricCounters {
-    private static final ConcurrentMap<CounterMetric, Integer> metrics = new ConcurrentHashMap<>();
-    public static final ThreadLocal<ConcurrentMap<CounterMetric, Integer>> Counters =
-            new ThreadLocal<ConcurrentMap<CounterMetric, Integer>>() {
-                @Override protected ConcurrentMap<CounterMetric, Integer> initialValue() {
-                    return metrics;
+    private static ConcurrentMetricCounters instance;
+    private static ConcurrentMap<CounterMetric, Integer> metrics = new ConcurrentHashMap<>();
+    public static ReentrantLock lock = new ReentrantLock();
+
+    private ConcurrentMetricCounters(){}
+
+    public static ConcurrentMetricCounters getInstance(){
+        if(instance == null){
```

Comment:
This top level if can probably be removed

Reply:
This is there for performance reasons. If this method is called often, `synchronized` can have some per implications. I don't expect this perf improvement to have huge consequences but still.